conbench/entities/benchmark_result.py

```diff
@@ -598,13 +599,40 @@ def ui_mean_and_uncertainty(values: List[float], unit: str) -> str:
     return f"({mean_uncertainty_str}) {unit}"
 
 
+def validate_augment_unit_string(u: str) -> str:
+    """
+    Raise BenchmarkResultValidationError for invalid unit string.
+
+    Return augmented unit string, to be inserted into database.
+    """
+    if u == "b/s":
+        # Rewrite short variant here for a legacy client, see
+        # https://github.com/conbench/conbench/issues/1335
+        # We might need/want to do a database migration where we rewrite
+        # b/s to B/s.
+        u = "B/s"
+
+    if u not in conbench.units.KNOWN_UNITS:
+        raise BenchmarkResultValidationError(
+            f"invalid unit string `{u}`, pick one of: {conbench.units.KNOWN_UNIT_SYMBOLS_STR}"
+        )
+
+    return u
+
+
 def validate_and_aggregate_samples(stats_usergiven: Any):
     """
     Raises BenchmarkResultValidationError upon logical inconsistencies.
 
+    `stats_usergiven` is deserialized JSON, validated against
+    `BenchmarkResultStatsSchema(marshmallow.Schema)`.
+
     Only run this for the 'success' case, i.e. when the input is a list longer
     than zero, and each item is a number (not math.nan).
 
+    Validate the user-given unit string also only in case of success, i.e.
+    allow for 'bad units' to be submitted with 'errored' results.
```

Comment:
Interesting decision here I needed to make: _when_  really to validate the unit string. We allow for submitting messy erroneous results, and so I went with that philosophy.